pkg/scaffold/api.go

```diff
@@ -269,14 +279,25 @@ func (api *API) scaffoldV2() error {
 	return nil
 }
 
-// Since we support single group only in v2 scaffolding, validate if resource
-// being created belongs to existing group.
-func (api *API) validateResourceGroup(r *resource.Resource) error {
+// isGroupAllowed will check if the group is == the group used before
+// and not allow new groups if the project is not enabled to use multigroup layout
+func (api *API) isGroupAllowed(r *resource.Resource) bool {
 	for _, existingGroup := range api.project.ResourceGroups() {
-		if !strings.EqualFold(r.Group, existingGroup) {
-			return fmt.Errorf("group '%s' is not same as existing group '%s'. Multiple groups are not supported yet.", r.Group, existingGroup)
+		if !strings.EqualFold(r.Group, existingGroup) && !api.project.MultiGroup {
+			return false
 		}
 	}
+	return true
+}
```

Comment:
I do like this approach, but you should only check once for multigroup at the start (short-circuit) to avoid checking in every loop iteration.

```
// isGroupAllowed will check if the group is == the group used before
// and not allow new groups if the project is not enabled to use multigroup layout
func (api *API) isGroupAllowed(r *resource.Resource) bool {
	// Groups are always allowed in multi-group projects
	if api.project.Multigroup {
		return true
	}
	for _, existingGroup := range api.project.ResourceGroups() {
		if !strings.EqualFold(r.Group, existingGroup) {
			return false
		}
	}
	return true
}
```

Reply:
good suggestion